tools/grub/Dockerfile

```diff
@@ -0,0 +1,69 @@
+FROM linuxkit/alpine:3683c9a66cd4da40bd7d6c7da599b2dcd738b559 AS grub-build
+RUN apk add \
+  automake \
+  make \
+  bison \
+  coreutils \
+  gettext \
+  gettext-dev \
+  flex \
+  gcc \
+  git \
+  libtool \
+  libc-dev \
+  linux-headers \
+  patch \
+  pkgconfig \
+  python3 \
+  autoconf
+
+# because python is not available
+RUN ln -s python3 /usr/bin/python
+
+ENV GRUB_MODULES="ext2 iso9660 gzio linux acpi normal cpio crypto disk boot crc64 part_gpt tftp xzio xfs video"
+ENV GRUB_COMMIT=5bc41db756c5b342aa22fc95508bde105e05a095
+ENV SRCDIR=/src
+
+# separate steps for better caching
+RUN mkdir /grub-lib ${SRCDIR} && \
+  set -e && \
+  cd ${SRCDIR} && \
+  git clone git://git.savannah.gnu.org/grub.git && \
+  cd grub && \
+  git checkout -b grub-build ${GRUB_COMMIT}
+
+WORKDIR ${SRCDIR}/grub
+
+# this is long and slow, so keep separate step
+RUN ./bootstrap
+
+# run these in separate steps for caching. 
+# determine uname -m in each step, since we want to replace it with a single cross-build when we get it working
+RUN  \
+  arch=$(uname -m); \
+  mkdir ${SRCDIR}/grub-${arch}; \
+  cd ${SRCDIR}/grub-${arch} ; \
+  ${SRCDIR}/grub/configure --libdir=/grub-lib --target=${arch} --with-platform=efi CFLAGS="-Os -Wno-unused-value" 
+
+RUN \
+  arch=$(uname -m); \
+  cd ${SRCDIR}/grub-${arch}; make -j "$(getconf _NPROCESSORS_ONLN)" && \
+  make install
+
+RUN \
+  arch=$(uname -m); \
+  [ "$arch" = "aarch64" ] && arch=arm64 ; \
+  case "$arch" in \
+  x86_64) \
+    outfile=BOOTX64.EFI; \
+    ;; \
+  arm64) \
```

Comment:
you can do `arm64,aarch64)` here its a bit cleaner.

Reply:
Doh! That is far simpler. Too much time in the gnu toolchain and you forget the incredibly simple sh things. Updating.

Reply:
A little cleaner, with `outarch`.